packages/store/src/decorators/select/select.ts

```diff
@@ -0,0 +1,50 @@
+import { Observable } from 'rxjs';
+
+import { SELECT_META_KEY } from '../../symbols';
+import { createSelect, createSelector } from './symbols';
+import { removeDollarAtTheEnd } from '../../internal/internals';
+import { CONFIG_MESSAGES, VALIDATION_CODE } from '../../configs/messages.config';
+
+/**
+ * Decorator for selecting a slice of state from the store.
+ */
+export function Select(selectorOrFeature?: any, ...paths: string[]): PropertyDecorator {
+  return function(target: any, key: string | symbol): void {
+    if (!Object.isExtensible(target)) {
+      throw new Error(CONFIG_MESSAGES[VALIDATION_CODE.SELECT_CLASS_NOT_EXTENSIBLE]());
+    }
+
+    const name: string = key.toString();
+    const selectorId: unique symbol = Symbol(`${SELECT_META_KEY}__${name}`);
+    selectorOrFeature = !selectorOrFeature ? removeDollarAtTheEnd(name) : selectorOrFeature;
+
+    Object.defineProperty(target, selectorId, {
+      writable: true,
+      enumerable: false,
+      configurable: true
+    });
+
+    Object.defineProperty(target, name, {
+      enumerable: true,
+      configurable: true,
+      get: function() {
```

Comment:
```ts
Object.defineProperty(target, name, {
  enumerable: true,
  configurable: true,
  get() {
    if (Object.isFrozen(target)) {
      throw new Error(CONFIG_MESSAGES[VALIDATION_CODE.SELECT_NOT_CONNECTED]());
    }

    return (
      this[selectorId] ||
      (this[selectorId] = createSelect.apply(this, [
        createSelector(
          selectorOrFeature,
          paths
        )
      ]))
    );
  }
});
```

Reply:
what is prefer 

```ts
 this[selectorId] ||
      (this[selectorId] = ...)
```

instead

```ts
let stream: Observable<any> | null = this[selectorId];

if (!stream) {
   stream = this[selectorId] = createSelect.apply(this, [
        createSelector(
           selectorOrFeature,
             paths
         )
   ]);
}

return stream;
```

I think this is more readable

Reply:
less code = win, also the problem is that `return this[selectorId] || ...` will always return the cached value first, while `if (!stream)` condition is not necessary to be called every time. Thoughts?

Reply:
Maybe then it’s worth writing a comment in the code then why do we do this

Reply:
Do we write comments for ourselves? I guess we boss understand this code :DD

Reply:
But other people can read our code.

Reply:
There is a concept of "self-documenting code". I agree with documenting dark parts of any code, too much comments is also bad, sounds doubtful. So this code:

```ts
this[selectorId] ||
      (this[selectorId] = ...)
```

is self-documenting IMO, but it's up to you.

Reply:
done